### ISSUE TITLE
Account naming mobile app

### ISSUE THREAD

**Issue by mnuky** [trezor/trezor-suite#10352]
## TL;DR
- we do not display setion headers (based on account type) in the account list in the end

## Full description

Current implementation of naming accounts does not correspond with account naming in Suite and FW and it may confuse user. 

Adjust the naming and design according these rules:
On My Asset page
- Each coin / Account type has its own container
- Coins with account types are sorted to separated under separated containers according to account type 
- Naming rules: each account type has its own naming and numbering of accounts

**Example:**
Bitcoin `#1` + badge `SegWit`
Bitcoin `#2` + badge `SegWit`

Bitcoin `#1` + badge `Taproot`
Bitcoin `#2` + badge `Taproot`

Bitcoin `#1`+ badge `Legacy SegWit`
Bitcoin `#2` + badge `Legacy SegWit`

Bitcoin `#1` + badge `Legacy`
Bitcoin `#2` + badge `Legacy`

Ethereum `#1`
Ethereum `#2`

Litecoin `#1` + badge `SegWit`
Litecoin `#2` + badge `SegWit`

Litecoin `#1` + badge `Legacy SegWit`
Litecoin `#2` + badge `Legacy SegWit`

Cardano `#1` (no badge for default account)
Cardano `#2` (no badge for default account)

Cardano `#1` + badge `Legacy`
Cardano `#2` + badge `Legacy`

Cardano `#1` + badge `Ledger`
Cardano `#2`+ badge `Ledger`

- These account names are reflected in header (breadcrumb navigation)
- Account types badge stays visible for coins which have account types
Default accounts are:
BTC: SegWit
LTC: SegWit
BTC Gold: Legacy SegWit
DigiByte: Legacy SegWit
Vertcoin: SegWit

For Cardano:
Default account type does not have badge
Legacy account type has badge `Legacy`
Ledger account type has badge `Ledger`

**Comment by bosomt** [comment]
**QA OK**

 24.01.1 (681)
e089477 - [develop]
